src/Development/Shake/Internal/Args.hs

```diff
@@ -179,7 +188,8 @@ shakeArgsOptionsWith baseOpts userOptions rules = do
                 (if null baseOpts2 then [] else "" : (if null userOptions then "Options:" else "Standard options:") : showOptDescr baseOpts2) ++
                 (if null userOptions then [] else "" : "Extra options:" : showOptDescr userOptions) ++
                 (if null changes then [] else "" : "Changed ShakeOptions:" : changes) ++
-                (if null targets then [] else "" : "Targets:" : targets)
+                (if null targets then [] else "" : "Targets:" : targets) ++
+                (if null helpSuffix then [] else "" : "User information:" : [helpSuffix])
```

Comment:
I wouldn't bother prepending `User information` - the user can add that with the suffix if they want.